### ISSUE TITLE
Getting "io/console on JRuby shells out to stty for most operations" on Jruby 9.3 with JDK17

### ISSUE THREAD

**Issue by aniljaiswal** [jruby/jruby#6853]
I'm running Jruby 9.3 on JDK17 locally on my MacBook (m1 chip) without any issues for a week now. But when I setup a docker image to do this, the container fails to run the rails console.

**Dockerfile**
```dockerfile
FROM openjdk:17-jdk-buster

RUN apt-get update && apt-get install -y build-essential git shared-mime-info libc6-dev --no-install-recommends && rm -rf /var/lib/apt/lists/*

ENV JRUBY_VERSION 9.3.0.0
ENV JRUBY_SHA256 2dc1f85936d3ff3adc20d90e5f4894499c585a7ea5fedec67154e2f9ecb1bc9b
RUN mkdir /opt/jruby \
  && curl -fSL https://repo1.maven.org/maven2/org/jruby/jruby-dist/${JRUBY_VERSION}/jruby-dist-${JRUBY_VERSION}-bin.tar.gz -o /tmp/jruby.tar.gz \
  && echo "$JRUBY_SHA256 /tmp/jruby.tar.gz" | sha256sum -c - \
  && tar -zx --strip-components=1 -f /tmp/jruby.tar.gz -C /opt/jruby \
  && rm /tmp/jruby.tar.gz \
  && update-alternatives --install /usr/local/bin/ruby ruby /opt/jruby/bin/jruby 1
ENV PATH /opt/jruby/bin:$PATH

# skip installing gem documentation
RUN mkdir -p /opt/jruby/etc \
       && { \
               echo 'install: --no-document'; \
               echo 'update: --no-document'; \
       } >> /opt/jruby/etc/gemrc

RUN gem install bundler rake net-telnet xmlrpc

# don't create ".bundle" in all our apps
ENV GEM_HOME /usr/local/bundle
ENV BUNDLE_SILENCE_ROOT_WARNING=1 \
       BUNDLE_APP_CONFIG="$GEM_HOME"
ENV PATH $GEM_HOME/bin:$PATH
# adjust permissions of a few directories for running "gem install" as an arbitrary user
RUN mkdir -p "$GEM_HOME" && chmod 777 "$GEM_HOME"

CMD ["irb"]
```

**Docker compose**
```yml
version: '3'

services:
  api:
    build: .
    stdin_open: true
    tty: true
    command: /bin/bash
    ports:
      - '3000:3000'
    volumes:
      - '.:/app'
    env_file:
      - '.env'
```

**Environment Information**
  
Provide at least:
* JRuby version: `jruby 9.3.0.0 (2.6.8) 2021-09-17 85c20e780f OpenJDK 64-Bit Server VM 17+35-2724 on 17+35-2724 +jit [linux-aarch64]`
* Operating system and platform: `Linux 955e16e092aa 5.10.47-linuxkit #1 SMP PREEMPT Sat Jul 3 21:50:16 UTC 2021 aarch64 GNU/Linux`
* Running on Docker inside Apple Macbook (M1 Chip)

Other relevant info you may wish to add:
* Application/framework version:  Rails 6.0.2
* Environment variables: Just database config

**Expected Behavior**
Rails console should open without any errors/warnings.

**Actual Behavior**
Running `rails c` just outputs `io/console on JRuby shells out to stty for most operations` and gets stuck.

Screenshot attached
<img width="465" alt="Screenshot 2021-09-24 at 4 31 13 PM" src="https://user-images.githubusercontent.com/43805107/134664187-491ef43d-15b2-4df9-a1de-bad7664adb01.png">

**Comment by aniljaiswal** [comment]
I tried running it 2-3 times and now it's working.

**Comment by aniljaiswal** [comment]
Reopening it as the issue is happening again. I have tried this with both 9.3 and 9.2.x versions. The issue happens only on ARM64 architecture (using Apple M1 chip).

**Comment by aniljaiswal** [comment]
@headius

**Comment by headius** [comment]
@aniljaiswal This is likely due to ongoing issues getting full M1 support and loading the jffi native library under strict macOS signing requirements.

You may have better luck, albeit reduced performance, by using an x86_64 JDK for now. I want to get M1 working but it requires some work in C that is a bit challenging.

**Comment by headius** [comment]
Likely fixed by jnr/jnr-posix#173 which has merged into all branches. Marking as fixed in 9.3, which is the release that will officially support Apple Silicon.

**Comment by headius** [comment]
Fixed by #6985 for 9.2 and #6986 for 9.3 and beyond.
